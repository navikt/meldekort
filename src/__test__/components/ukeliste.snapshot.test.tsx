import * as React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderWrapper from '../utils/providerWrapper';
import Ukeliste from '../../app/components/meldekortdetaljer/ukevisning/ukeliste';
import {
  KortStatus,
  KortType,
  Meldegruppe,
  Meldekort,
  MeldekortDag,
  Meldeperiode,
} from '../../app/types/meldekort';

Enzyme.configure({ adapter: new Adapter() });

it('Ukeliste', () => {
  const tree = shallow(
    <ProviderWrapper>
      <Ukeliste aktivtMeldekort={mk} erAap={true} meldekortDager={mkDager} ukeNr={1} />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});

const dateString = '2019-01-01T10:00:00';

const meldeperiode: Meldeperiode = {
  til: new Date(dateString),
  fra: new Date(dateString),
  kanKortSendes: true,
  kortKanSendesFra: new Date(dateString),
  periodeKode: '201901',
};

const mk: Meldekort = {
  meldekortId: 1,
  kortType: KortType.ELEKTRONISK,
  meldeperiode: meldeperiode,
  meldegruppe: Meldegruppe.DAGP,
  kortStatus: KortStatus.OPPRE,
  bruttoBelop: 0,
  mottattDato: new Date(dateString),
  korrigerbart: true,
};

const mkDager: MeldekortDag[] = [
  {
    dag: 1,
    arbeidetTimerSum: 7.5,
    syk: false,
    annetFravaer: true,
    kurs: false,
  },
];
