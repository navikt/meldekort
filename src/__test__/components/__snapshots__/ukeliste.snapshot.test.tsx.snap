// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ukeliste 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(IntlProvider)
    defaultLocale="nb"
    locale="nb"
  >
    <Ukeliste
      aktivtMeldekort={
        Object {
          "bruttoBelop": 0,
          "korrigerbart": true,
          "kortStatus": "OPPRE",
          "kortType": "ELEKTRONISK",
          "meldegruppe": "DAGP",
          "meldekortId": 1,
          "meldeperiode": Object {
            "fra": 2019-01-01T10:00:00.000Z,
            "kanKortSendes": true,
            "kortKanSendesFra": 2019-01-01T10:00:00.000Z,
            "periodeKode": "201901",
            "til": 2019-01-01T10:00:00.000Z,
          },
          "mottattDato": 2019-01-01T10:00:00.000Z,
        }
      }
      meldekortDager={
        Array [
          Object {
            "annetFravaer": true,
            "arbeidetTimerSum": 7.5,
            "dag": 1,
            "kurs": false,
            "syk": false,
          },
        ]
      }
      typeYtelsePostfix="-AAP"
      ukeNr={1}
    />
  </Connect(IntlProvider)>
</Provider>
`;
